packages/@aws-cdk/aws-pipes-targets-alpha/test/integ.lambda.ts

```diff
@@ -0,0 +1,94 @@
+import * as path from 'path';
+import {
+  IPipe,
+  ISource,
+  Pipe,
+  SourceConfig,
+} from '@aws-cdk/aws-pipes-alpha';
+import { ExpectedResult, IntegTest } from '@aws-cdk/integ-tests-alpha';
+import * as cdk from 'aws-cdk-lib';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import { LambdaFunction } from '../lib';
+
+/**
+ * Stack verification steps:
+ * 1. The message 'Sunny' is sent to the queue 'SourceQueue', which triggers the Lambda Function from the pipe.
+ * 2. The Lambda Function adds the tag:
+ *    Key: Weather
+ *    Value: Sunny
+ * 3. The assertion verifies that the expected tag is created by calling listTags on the Lambda Function.
+ */
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'aws-cdk-pipes-lambda-target');
+const sourceQueue = new cdk.aws_sqs.Queue(stack, 'SourceQueue');
+
+const functionName = 'TestCdkPipesTargetLambdaFunction';
+const targetFunction = new lambda.Function(stack, 'TargetLambdaFunction', {
+  code: lambda.AssetCode.fromAsset(
+    path.join(__dirname, 'integ.lambda.handler'),
+  ),
+  handler: 'handler.handler',
+  functionName,
+  runtime: lambda.Runtime.NODEJS_LATEST,
+  memorySize: 512,
+});
+targetFunction.addToRolePolicy(
+  new iam.PolicyStatement(
+    new iam.PolicyStatement({
+      actions: ['lambda:TagResource'],
+      resources: ['*'],
+    }),
+  ),
+);
+
+class TestSource implements ISource {
+  sourceArn: string;
+  sourceParameters = undefined;
+  constructor(private readonly queue: cdk.aws_sqs.Queue) {
+    this.queue = queue;
+    this.sourceArn = queue.queueArn;
+  }
+  bind(_pipe: IPipe): SourceConfig {
+    return {
+      sourceParameters: this.sourceParameters,
+    };
+  }
+  grantRead(pipeRole: cdk.aws_iam.IRole): void {
+    this.queue.grantConsumeMessages(pipeRole);
+  }
+}
+
+new Pipe(stack, 'Pipe', {
+  source: new TestSource(sourceQueue),
+  target: new LambdaFunction(targetFunction, {}),
+});
+
+const test = new IntegTest(app, 'integtest-pipe-target-lambda', {
+  testCases: [stack],
+});
+
+const sunny = 'Sunny';
```

Comment:
Personally I would use a random value to make it's your change that we assert. If you develop the integ test and do not destroy the stack it could happen that the update of the tag fails but still has the value from the previous run.

Reply:
Did that